deployer/src/deployment/mod.rs

```diff
@@ -145,12 +149,15 @@ where
             storage_manager.clone(),
         ));
 
-        DeploymentManager {
-            queue_send,
-            run_send,
-            runtime_manager,
-            storage_manager,
-        }
+        (
+            set,
```

Comment:
Can we attach these to the deployment manager?

Reply:
If we attach it to the deployer, then we have to wrap it in an `Arc` and a `Mutex`. I tried to avoid that (no particular reason actually, now that I think about it) but it might be better to wrap it and keep the `JoinSet` with the manager.